service/src/main/java/bio/terra/pipelines/app/controller/PipelinesApiController.java

```diff
@@ -183,56 +177,69 @@ public ResponseEntity<ApiCreateJobResponse> createJob(
+    PipelineApiUtils.validatePipelineName(pipelineName, logger);
 
-    return new ResponseEntity<>(result, HttpStatus.OK);
-  }
-
-  @Override
-  public ResponseEntity<ApiCreateJobResponse> getPipelineJobResult(
-      @PathVariable("pipelineName") String pipelineName, @PathVariable("jobId") UUID jobId) {
-    final SamUser userRequest = getAuthenticatedInfo();
-    String userId = userRequest.getSubjectId();
-    PipelinesEnum validatedPipelineId = PipelineApiUtils.validatePipelineName(pipelineName, logger);
+    PipelineRun pipelineRun = pipelineRunsService.getPipelineRun(jobId, userId);
+    if (pipelineRun == null) {
+      throw new NotFoundException("Pipeline run %s not found".formatted(jobId));
+    }
 
-    JobApiUtils.AsyncJobResult<String> jobResult =
-        jobService.retrieveAsyncJobResult(jobId, userId, validatedPipelineId, String.class, null);
+    ApiCreatePipelineRunResponse runResponse = pipelineRunToApi(pipelineRun);
 
-    ApiCreateJobResponse response =
-        new ApiCreateJobResponse()
-            .jobReport(jobResult.getJobReport())
-            .errorReport(jobResult.getApiErrorReport())
-            .pipelineOutput(jobResult.getResult()); // this is null unless the job succeeded
+    return new ResponseEntity<>(runResponse, getAsyncResponseCode(runResponse.getJobReport()));
+  }
 
-    return new ResponseEntity<>(response, getAsyncResponseCode(response.getJobReport()));
+  /**
+   * Converts a PipelineRun to an ApiCreatePipelineRunResponse. If the PipelineRun has completed
+   * successfully (isSuccess is true), we know there are no errors to retrieve and all the
+   * information needed to return to the user is available in the pipeline_runs table.
+   *
+   * <p>If the PipelineRun is not marked as a success, we retrieve the running and/or error
+   * information from Stairway.
+   *
+   * @param pipelineRun the PipelineRun to convert
+   * @return ApiCreatePipelineRunResponse
+   */
+  private ApiCreatePipelineRunResponse pipelineRunToApi(PipelineRun pipelineRun) {
+    if (Boolean.TRUE.equals(pipelineRun.getIsSuccess())) {
+      return new ApiCreatePipelineRunResponse()
+          .jobReport(
+              new ApiJobReport()
+                  .id(pipelineRun.getJobId().toString())
+                  .description(pipelineRun.getDescription())
+                  .status(ApiJobReport.StatusEnum.SUCCEEDED)
+                  .statusCode(HttpStatus.OK.value())
+                  .submitted(pipelineRun.getCreated().toString())
+                  .completed(pipelineRun.getUpdated().toString())
+                  .resultURL(pipelineRun.getResultUrl()))
+          .pipelineOutput(pipelineRun.getOutput());
+    } else {
+      JobApiUtils.AsyncJobResult<String> jobResult =
+          jobService.retrieveAsyncJobResult(
+              pipelineRun.getJobId(), pipelineRun.getUserId(), String.class, null);
+
+      return new ApiCreatePipelineRunResponse()
```

Comment:
the following will populate the submitted and completed fields with timestamps from stairway, not from the pipeline_runs table. this means that calling GET pipelines/imputation_beagle/result/{job_id} on a successful run will return different timestamps than calling GET jobs/{job_id}. are we ok with this or should we make them the same? they're off from each other by milliseconds.

Reply:
So for completedTime if its running it should be null.  If its errored then the user will always see the same time.  If its successful then the user will alway see the same time.  completedTime seems like it'll always be consistent from the users perspective.

For start time the only time it'll change is when the start time they see when its running/failed and the start time they see when its a success.  Its not the best feeling in the world but I'm not sure its worth the effort to bring in the pipeline_runs start time in the `else` part of the code.